src/sentry/utils/cache.py

```diff
@@ -8,43 +8,34 @@
 from __future__ import absolute_import, print_function
 
 import functools
+from functools32 import lru_cache
 
 from django.core.cache import cache
 
 default_cache = cache
 
 
-class memoize(object):
+def memoize(func):
     """
-    Memoize the result of a property call.
+    A decorator that returns a cached property getter. This should work the
+    same as @property but the result of the underlying function will be cached.
+    Uses lru_cache to store the cached property to avoid polluting the object's
+    namespace.
 
     >>> class A(object):
     >>>     @memoize
     >>>     def func(self):
     >>>         return 'foo'
     """
+    @property
+    @lru_cache(maxsize=10)
+    def prop(*args):
```

Comment:
Do you need to support keyword args?

Reply:
nah, is just a getter, so actually the only possible argument is `self`, so I should just make it that.